cloud-function/calling-skygear-api/js.md

```diff
@@ -0,0 +1,160 @@
+---
+title: Calling Skygear API
+---
+
+[[toc]]
+
+
+## Skygear Container
+
+Besides using the SDKs to interact with Skygear,
+you can also call the Skygear APIs directly using the Skygear
+`Container` from the cloud code.
+
+```javascript
+const container = new skygearCloud.CloudCodeContainer();
+container.apiKey = 'your-api-key';
+container.endPoint = skygearCloud.settings.skygearEndpoint + '/';
+```
+
+You can create Skygear Container object by `CloudCodeContainer()` from `skygearCloud`. And configure it with `apiKey`, `endPoint` and `userId`.
+
+Query can be done similar to JS SDK.
+
+As an example, you can perform a query through Skygear with the following:
+
+```javascript
+const container = new skygearCloud.CloudCodeContainer();
+container.apiKey = skygearCloud.settings.masterKey;
+container.endPoint = skygearCloud.settings.skygearEndpoint + '/';
+container.asUserId = 'admin';
+const TaskRecord = skygear.Record.extend('task');
+const query = new skygear.Query(TaskRecord);
+query.equalTo('_id', 'cdfc7bb4-afd3-464c-a430-c9564c2202cf');
+container.publicDB.query(query).then((records) => {
+	console.log(records);
+});
+```
+
+::: caution
+
+**Caution:** By using the app's API key, the `SkygearContainer`
+does not have an authenticated user associated.
+Any actions or queries made are assumed to be made by an
+unauthenticated user, and are subject to access control set
+to the public. Therefore you cannot save or alter records
+with this API key. To have a user-aware `SkygearContainer`
+for such operations, you need to use the master key, explained below.
+
+:::
+
+### Using the master key
+
+A master key is a special key which allows you to perform operations
+that would not be possible using the normal API key.
+A typical use case in the cloud code is to impersonate a user for
+creating, altering or querying records.
+
+You can find your master key in your
+[app settings in the Skygear portal][portal-app-settings],
+or obtain your master key through the cloud code by:
+
+```javascript
+masterKey = skygearCloud.settings.masterKey;
```

Comment:
Add example on how to use the master key

Reply:
yea, there is one example on demonstrating saving a record on behalf of the authenticated user